jwst/picture_frame/picture_frame_step.py

```diff
@@ -0,0 +1,112 @@
+
+    class_alias = "picture_frame"
+
+    spec = """
+        mask_science_regions = boolean(default=True)  # Mask known science regions.
+        n_sigma = float(default=2.0)  # Clipping level for non-background signal.
+        save_mask = boolean(default=False)  # Save the created background mask.
+        save_correction = boolean(default=False)  # Save the thermal correction data.
+        skip = boolean(default=True)  # By default, skip the step.
+    """  # noqa: E501
+
+    reference_file_types = ["pictureframe"]
+
+    def process(self, input_data):
+        """
+        Scale and subtract the thermal "picture frame" effect from a ramp data set.
+
+        First, a draft rate image is created if needed from the input ramp data.
+        A background mask is created from the rate image by masking any known science
+        regions, then iteratively sigma-clipping.  Then, the median levels are computed
+        from background data in a central region and from the edges of each group image.
+        These levels are used to scale and offset the correction image in the
+        ``pictureframe`` reference file, then the scaled image is subtracted from the
+        input data in each group.
+
+        Input data is expected to be a NIRSpec FULL frame ramp file (RampModel),
+        in between jump and ramp fitting steps (before 1/f noise cleaning),
+        or a rate file (ImageModel or CubeModel).
+
+        Alternately, a rate image or rateints cube may be provided.  In that case,
+        no draft rate image is necessary, and the correction is directly applied
+        to the rate image.
+
+        Parameters
+        ----------
+        input_data : str or `~stdatamodels.jwst.datamodels.RampModel` \
+                     or `~stdatamodels.jwst.datamodels.ImageModel` \
+                     or `~stdatamodels.jwst.datamodels.CubeModel`
+            Filename or input datamodel to be corrected. Must be NIRSpec full-frame.
+
+        Returns
+        -------
+        output_model : `~stdatamodels.jwst.datamodels.RampModel` \
+                       or `~stdatamodels.jwst.datamodels.ImageModel` \
+                       or `~stdatamodels.jwst.datamodels.CubeModel`
+            The corrected datamodel, matching the type of the input.
+        """
+        # Open the input data model
+        # Note: this step can be run either in stage 1 on ramp-type
+        # data or in stage 2 on rate-type data, so it's important
+        # not to provide a specific datamodel to prepare_output.
+        output_model = self.prepare_output(input_data)
+
+        pictureframe_file = self.get_reference_file(output_model, "pictureframe")
+        exp_type = output_model.meta.exposure.type
+        if pictureframe_file == "N/A":
+            log.warning(f"Picture frame correction is not available for exposure type {exp_type}.")
+            status = "SKIPPED"
+            mask_model = None
+            correction_model = None
```

Comment:
instead of setting these to None and handling that case below, could you just return here?